src/OmniSharp.Roslyn.CSharp/Services/Intellisense/CompletionItemExtensions.cs

```diff
@@ -18,56 +19,60 @@ internal static class CompletionItemExtensions
         private static MethodInfo _getSymbolsAsync;
+        private static readonly PropertyInfo _getProviderName;
 
         static CompletionItemExtensions()
         {
-            var symbolCompletionItemType = typeof(CompletionItem).GetTypeInfo().Assembly.GetType(SymbolCompletionItem);
-            _getSymbolsAsync = symbolCompletionItemType.GetMethod(GetSymbolsAsync, BindingFlags.Public | BindingFlags.Static);
+            _symbolCompletionItemType = typeof(CompletionItem).GetTypeInfo().Assembly.GetType(SymbolCompletionItem);
+            _getSymbolsAsync = _symbolCompletionItemType.GetMethod(GetSymbolsAsync, BindingFlags.Public | BindingFlags.Static);
+
+            _getProviderName = typeof(CompletionItem).GetProperty(ProviderName, BindingFlags.NonPublic | BindingFlags.Instance);
+        }
+
+        private static string GetProviderName(CompletionItem item)
+        {
+            return (string)_getProviderName.GetValue(item);
         }
 
         public static bool IsObjectCreationCompletionItem(this CompletionItem item)
         {
-            var properties = item.Properties;
-            return properties.TryGetValue(Provider, out var provider) && provider == ObjectCreationCompletionProvider;
+            return GetProviderName(item) == ObjectCreationCompletionProvider;
         }
 
         public static async Task<IEnumerable<ISymbol>> GetCompletionSymbolsAsync(this CompletionItem completionItem, IEnumerable<ISymbol> recommendedSymbols, Document document)
         {
-            var properties = completionItem.Properties;
-
-            // for SymbolCompletionProvider, use the logic of extracting information from recommended symbols
-            if (properties.TryGetValue(Provider, out var provider) && provider == SymbolCompletionProvider)
+            if (completionItem.GetType() == _symbolCompletionItemType)
             {
-                return recommendedSymbols.Where(x => x.Name == properties[SymbolName] && (int)x.Kind == int.Parse(properties[SymbolKind])).Distinct();
-            }
-
-            // if the completion provider encoded symbols into Properties, we can return them
-            if (properties.ContainsKey(Symbols))
-            {
-                // the API to decode symbols is not public at the moment
-                // http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Completion/Providers/SymbolCompletionItem.cs,93
                 var decodedSymbolsTask = _getSymbolsAsync.InvokeStatic<Task<ImmutableArray<ISymbol>>>(new object[] { completionItem, document, default(CancellationToken) });
                 if (decodedSymbolsTask != null)
                 {
                     return await decodedSymbolsTask;
                 }
             }
 
+            var properties = completionItem.Properties;
+
+            // if the completion provider encoded symbols into Properties, we can return them
+            if (properties.ContainsKey(SymbolName) && properties.ContainsKey(SymbolName))
```

Comment:
I read changelog and saw that line. `properties.ContainsKey(SymbolName)` - why this code is duplicated? Maybe second check must be `properties.ContainsKey(SymbolKind)` ?

Reply:
yes, it's a typo thanks for finding this 👍I opened the fix here https://github.com/OmniSharp/omnisharp-roslyn/pull/1491